app/uk/gov/hmrc/agentsubscription/repository/SubscriptionJourneyRepository.scala

```diff
@@ -88,11 +101,136 @@ class SubscriptionJourneyRepositoryImpl @Inject() (mongo: MongoComponent)(implic
+      contactTradingAddress = contactTradingAddressData.contactTradingAddress.map(decryptBusinessAddress)
+    )
+
+  def encryptBusinessAddress(businessAddress: BusinessAddress): BusinessAddress =
+    businessAddress.copy(
+      addressLine1 = crypto.encrypt(PlainText(businessAddress.addressLine1)).value,
+      addressLine2 = businessAddress.addressLine2.map { f: String => crypto.encrypt(PlainText(f)).value },
+      addressLine3 = businessAddress.addressLine3.map { f: String => crypto.encrypt(PlainText(f)).value },
+      addressLine4 = businessAddress.addressLine4.map { f: String => crypto.encrypt(PlainText(f)).value },
+      postalCode = businessAddress.postalCode.map { f: String => crypto.encrypt(PlainText(f)).value },
+      countryCode = crypto.encrypt(PlainText(businessAddress.countryCode)).value
+    )
+
+  private def decryptBusinessAddress(businessAddress: BusinessAddress): BusinessAddress =
+    businessAddress.copy(
+      addressLine1 = crypto.decrypt(Crypted(businessAddress.addressLine1)).value,
+      addressLine2 = businessAddress.addressLine2.map { f: String => crypto.decrypt(Crypted(f)).value },
+      addressLine3 = businessAddress.addressLine3.map { f: String => crypto.decrypt(Crypted(f)).value },
+      addressLine4 = businessAddress.addressLine4.map { f: String => crypto.decrypt(Crypted(f)).value },
+      postalCode = businessAddress.postalCode.map { f: String => crypto.decrypt(Crypted(f)).value },
+      countryCode = crypto.decrypt(Crypted(businessAddress.countryCode)).value
+    )
+
+  def encryptRecord(record: SubscriptionJourneyRecord): SubscriptionJourneyRecord =
+    record.copy(
+      encrypted = Some(true),
+      businessDetails = encryptBusinessDetails(record.businessDetails),
+      contactEmailData = record.contactEmailData.map(encryptContactEmailData),
+      contactTradingNameData = record.contactTradingNameData.map(encryptContactTradingNameData),
+      contactTelephoneData = record.contactTelephoneData.map(encryptContactTelephoneData),
+      contactTradingAddressData = record.contactTradingAddressData.map(encryptContactTradingAddressData),
+      verifiedEmails = record.verifiedEmails.map { f: String => crypto.encrypt(PlainText(f)).value }
+    )
+
+  private def decryptRecord(record: SubscriptionJourneyRecord): SubscriptionJourneyRecord =
+    if (record.encrypted.contains(true)) {
+      record.copy(
+        businessDetails = decryptBusinessDetails(record.businessDetails),
+        contactEmailData = record.contactEmailData.map(decryptContactEmailData),
+        contactTradingNameData = record.contactTradingNameData.map(decryptContactTradingNameData),
+        contactTelephoneData = record.contactTelephoneData.map(decryptContactTelephoneData),
+        contactTradingAddressData = record.contactTradingAddressData.map(decryptContactTradingAddressData),
+        verifiedEmails = record.verifiedEmails.map { f: String => crypto.decrypt(Crypted(f)).value }
+      )
+    } else record
+
```

Comment:
I think these would be better as methods in companion objects to their respective models as this is not something the database layer needs to be concerned with really